src/math/vector.f90

```
@@ -174,6 +179,23 @@ function vector_size(v) result(s)
     s = v%n
   end function vector_size
 
+  !> Easy way to copy between host and device.
+  !! @param v vector to copy to/from device/host
+  !! @memdir direction to copy (HOST_TO_DEVICE or DEVICE_TO_HOST)
+  !! @sync whether the memcopy to be blocking or not
+  subroutine vector_copyto(v, memdir, sync)
+    class(vector_t), intent(inout) :: v
+    integer, intent(in) :: memdir
+    logical, intent(in) :: sync
+
+    if (NEKO_BCKND_DEVICE .eq. 1) then
+       call device_memcpy(v%x, v%x_d, v%n, &
+            memdir, sync)
+    end if
+
+  end subroutine vector_copyto
```

same as for `field.f90`, maybe throw an error here